apps/website/src/content/docs/themeprovider.mdx

```diff
@@ -0,0 +1,140 @@
+
+## Theming
+
+The `theme` prop can be set to either `"light"` or `"dark"`. This will automatically and performantly update the theme for the entire tree, by changing the values of [CSS variables](variables).
+
+```jsx
+<ThemeProvider theme='dark'>…</ThemeProvider>
+```
+
+This theme affects all components that use iTwinUI's CSS variables, including custom components and components from other packages.
+
+If you want to set the theme based on the user's operating system, you can use the `"os"` value. This relies on the [`prefers-color-scheme` query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) to be executed within client-side JavaScript, so it may cause a [flash of incorrect theme](https://css-tricks.com/flash-of-inaccurate-color-theme-fart/) when using server-side rendering.
+
+```jsx
+<ThemeProvider theme='os'>…</ThemeProvider>
+```
+
+If a `theme` value is not specified, it will default to [`"inherit"`](#inheritance), which will inherit the theme from its ancestors and fall back to `"light"`.
+
+### High contrast mode
+
+To enable higher contrast versions of light and dark themes, `themeOptions.highContrast` can be used. By default, the [`prefers-contrast` query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast) will be used to automatically enable high contrast themes if the user has a specified a preference for it in their operating system.
+
+```jsx
+<ThemeProvider
+  theme='dark'
+  themeOptions={{
+    highContrast: true,
+  }}
+/>
+```
+
+### Background
+
+By default, the topmost `ThemeProvider` in the tree will set the `background-color` to `var(--iui-color-background-backdrop)`, which is the recommended value for page background.
+
+If you want to explicitly control whether or not the backgrond is applied, you can override this behavior using `themeOptions.applyBackground`:
+
+```jsx
+<ThemeProvider
+  themeOptions={{
+    applyBackground: false, // or true
+  }}
+/>
+```
+
+## Scoping and nesting
+
+By default, `ThemeProvider` will only apply its theme to its descendants, i.e. components outside of the provider's subtree will not be affected.
+
+This is useful if you want to have multiple themes in your application, or if you want to use iTwinUI for only a portion of your application.
+
+`ThemeProvider`s can also be nested if you want to use a different theme for a portion of your application.
+
+```jsx
+<ThemeProvider theme='light'>
+  <Text>Uses light theme!</Text>
+
+  <ThemeProvider theme='dark'>
+    <Text>Uses dark theme!</Text>
+  </ThemeProvider>
+</ThemeProvider>
+```
+
+### Inheritance
+
+When using `theme="inherit"` (which is also the default value), `ThemeProvider` will inherit the theme from its ancestors, e.g. from a `ThemeProvider` higher up in the tree.
+
+The inherited theme can also come from an older version of iTwinUI or from standalone usage of [variables](variables). If no parent theme is found, it will fall back to `"light"`.
```

Comment:
we need to document standalone usage of itwinui-variables some more, but this is what i mean:

```html
<body data-iui-theme="dark">
```
```js
import '@itwin/itwinui-variables';
```

Reply:
yes, you are confused because it is not documented.

`data-iui-theme` is an implementation detail of `itwinui-react`. adding it without `itwinui-variables` will do nothing because `itwinui-react` styles are scoped.

only the variables themselves (`--iui-color-background`, etc) can be used within the `ThemeProvider` tree.

if you want to use it outside the tree (for example on `<html>` or `<body>`), then you need to use it ["standalone"](https://github.com/iTwin/iTwinUI/blob/main/packages/itwinui-variables/README.md)